feat: move agents to new tile

jor2 opened the pull request:
### Description

<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+{
+    "products": [
+      {
+        "name": "deploy-arch-ibm-scc-workload-protection-agent",
```
maybe go with `deploy-arch-ibm-scc-wp-agent`

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+    "products": [
+      {
+        "name": "deploy-arch-ibm-scc-workload-protection-agent",
+        "label": "Cloud automation for Security and Compliance Center Agents",
```
It is a bit long, but I think the correct working would be:
"Cloud automation for Security and Compliance Center Workload Protection agent"

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+          "terraform",
+          "solution"
+        ],
+        "short_description": "Creates and configures IBM Security and Compliance Center resources",
```
Configures Security and Compliance Center Workload Protection agents

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+          "solution"
+        ],
+        "short_description": "Creates and configures IBM Security and Compliance Center resources",
+        "long_description": "This architecture supports creating and configuring IBM Security and Compliance Center resources.",
```
This architecture supports creating and configuring IBM Security and Compliance Center Workload Protection agent to an IBM Cloud Kubernentes or Openshift cluster

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+        ],
+        "short_description": "Creates and configures IBM Security and Compliance Center resources",
+        "long_description": "This architecture supports creating and configuring IBM Security and Compliance Center resources.",
+        "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/blob/main/solutions/instances/README.md",
```
needs to change to new location

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+        "short_description": "Creates and configures IBM Security and Compliance Center resources",
+        "long_description": "This architecture supports creating and configuring IBM Security and Compliance Center resources.",
+        "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/blob/main/solutions/instances/README.md",
+        "offering_icon_url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-scc-da/main/images/scc-icon.svg",
```
suggest to move the icon too

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+        "flavors": [
+          {
+            "label": "Agents",
+            "name": "agents",
```
We no longer need to call it agents. Suggest to go with:

"label": "Standard",
"name": "standard",

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+          {
+            "label": "Agents",
+            "name": "agents",
+            "install_type": "extension",
```
change this to "fullstack" and add 
`"dependency_version_2": true,`

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+            "label": "Agents",
+            "name": "agents",
+            "install_type": "extension",
+            "working_directory": "solutions/agents",
```
I guess move the code to a `standard` directory to match the new proposed variation name

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+                "role_crns": [
+                  "crn:v1:bluemix:public:iam::::serviceRole:Manager",
+                  "crn:v1:bluemix:public:iam::::role:Editor"
+                ]
```
these permissions are not needed for agents

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,209 @@
+                {
+                  "diagram": {
+                    "caption": "Security and Compliance Center Workload Protection agent",
+                    "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-scc-da/main/reference-architecture/scc-wp-agent.svg",
```
update diagram link to this repo

ocofaigh reviewed (changes requested):
See comments. Can you also:
- add a `.catalog-onboard-pipeline.yaml` (Ill share the offering Id after next commit to this PR)
- update .releaserc to trigger the catalog pipeline

jor2 commented:
> See comments. Can you also:
> 
> * add a `.catalog-onboard-pipeline.yaml` (Ill share the offering Id after next commit to this PR)
> * update .releaserc to trigger the catalog pipeline

updated and added missing scripts

ocofaigh commented on .catalog-onboard-pipeline.yaml:
```diff
@@ -0,0 +1,13 @@
+  - name: deploy-arch-ibm-scc-wp-agent
+    kind: solution
+    catalog_id: 7df1e4ca-d54c-4fd0-82ce-3d13247308cd
+    offering_id: 9423f9bc-1290-4c71-a9ac-01898bfa7ccc
```
@jor2 change this to `f76f2ed3-eb98-4e3d-b147-360485fd8a36`

jor2 commented:
/run pipeline

ocofaigh commented on tests/pr_test.go:
```diff
@@ -2,16 +2,37 @@
 const basicExampleDir = "examples/basic"
 const secureExampleDir = "examples/secure"
+const agentFlavorDir = "solutions/agents"
+const agentsKubeconfigDir = "solutions/agents/kubeconfig"
```
this need to be updated to /solutions/standard. I think thats why the test failed

jor2 replied:
oh :(

jor2 commented:
/run pipeline

jor2 commented:
/run pipeline

jor2 commented:
/run pipeline

ocofaigh commented:
@jor2 you need to resolve conflicts before pipeline can be triggered

jor2 commented:
/run pipeline

jor2 commented:
/run pipeline

terraform-ibm-modules-ops commented:
:tada: This PR is included in version 1.4.0 :tada:

The release is available on:
- [GitHub release](https://github.com/terraform-ibm-modules/terraform-ibm-scc-workload-protection-agent/releases/tag/v1.4.0)
- `v1.4.0`

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
